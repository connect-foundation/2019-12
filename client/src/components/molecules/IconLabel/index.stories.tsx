import React from 'react';
import { text } from '@storybook/addon-knobs';

import IconLabel from '.';
import { FaCheck } from 'react-icons/fa';

export default {
  title: 'Molecules / IconLabel',
};

export const index: React.FC = () => (
  <IconLabel
    icon={<FaCheck size={'1.5rem'} />}
    labelContent={text('labelContent', '1인당 2개 구입 가능')}
  />
);
