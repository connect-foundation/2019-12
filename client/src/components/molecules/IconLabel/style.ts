import styled from 'styled-components';
import { theme } from 'styled-tools';

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  margin: 1rem 0rem;
`;

export const Label = styled.div`
  ${theme('fontStyle.subtitle2')}
  margin-left: 1rem;
`;
