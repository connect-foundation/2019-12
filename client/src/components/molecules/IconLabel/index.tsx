import React from 'react';

import * as S from './style';

export interface Props {
  icon: React.ReactNode;
  labelContent: string;
}

function IconLabel({ icon, labelContent }: Props): React.ReactElement {
  return (
    <S.Container>
      {icon}
      <S.Label>{labelContent}</S.Label>
    </S.Container>
  );
}

export default IconLabel;
