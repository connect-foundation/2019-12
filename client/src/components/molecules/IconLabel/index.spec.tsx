import React from 'react';
import { mount } from 'enzyme';

import IconLabel from '.';
import Check from 'assets/img/check-black.svg';

describe('Atom / IconBtn', () => {
  it('[SNAPSHOT] 렌더링', () => {
    // given
    const wrapper = mount(
      <IconLabel
        iconProps={{
          height: '1.5rem',
          alt: 'check',
          src: Check,
        }}
        labelContent={'1인당 2개 구입 가능'}
      />,
    );

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
