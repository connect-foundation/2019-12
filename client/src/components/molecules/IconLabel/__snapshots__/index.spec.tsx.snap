// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom / IconBtn [SNAPSHOT] 렌더링 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 1rem 0rem;
}

.c2 {
  margin-left: 1rem;
}

.c1 {
  height: 1.5rem;
}

<IconLabel
  iconProps={
    Object {
      "alt": "check",
      "height": "1.5rem",
      "src": "check-black.svg",
    }
  }
  labelContent="1인당 2개 구입 가능"
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  margin: 1rem 0rem;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <Icon
          alt="check"
          height="1.5rem"
          src="check-black.svg"
        >
          <styled.img
            alt="check"
            height="1.5rem"
            src="check-black.svg"
          >
            <StyledComponent
              alt="check"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htpNat",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  height: ",
                      [Function],
                      ";
  border-radius: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.img",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-htpNat",
                  "target": "img",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="1.5rem"
              src="check-black.svg"
            >
              <img
                alt="check"
                className="c1"
                height="1.5rem"
                src="check-black.svg"
              />
            </StyledComponent>
          </styled.img>
        </Icon>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "
  ",
                    [Function],
                    "
  margin-left: 1rem;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c2"
            >
              1인당 2개 구입 가능
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</IconLabel>
`;
